pkg/log/log_test.go

```diff
@@ -340,3 +340,85 @@ func TestNewAuditLogger(t *testing.T) {
 		}, ShouldPanic)
 	})
 }
+
+func TestCallerFunction(t *testing.T) {
```

Comment:
Let me see if I understand correctly:
1. The logger functioned correctly in cases such as `logger.Info().Msg("test message 1")`
2. In cases where other functions were called after event initialization, such as `logger.Info().Str("key1", "val2").Str("key2", "val2").Int("key3", 1).Msg("test message 1")`, the old code should have been counting an extra frame to be skipped for each of them in the Handle() method (in this case 3), but was not.
3. By moving the frame counting to newEvent(), the calls made between newEvent() and Handle() no longer matter.